### ISSUE TITLE
Use URL field in password data for matching

### ISSUE THREAD

**Issue by mbrunnen** [passff/passff#374]
The URL field in the password data should be, when present, the only value for URL matching quality. It seems right now, only the file name is taken in consideration. That leads, at least in my case, to a lot of false positives.

I think [`getItemQuality`](https://github.com/passff/passff/blob/master/src/modules/pass.js#L152) is the according function.

**Comment by tuxor1337** [comment]
Impossible: We cannot match on the url field or any other entry content, because this would require decrypting the whole password store (see also https://github.com/passff/passff/issues/360 and several other previous requests).

**Comment by mbrunnen** [comment]
oki thanks.

**Comment by sunaurus** [comment]
> We cannot match on the url field or any other entry content, because this would require decrypting the whole password store

@tuxor1337 can you go into a bit more detail about what exactly is the problem here? This is the behavior of `pass grep` anyway, which is built into pass itself, and if the behavior is not a problem for pass, why is it a problem for passff? 

If it's a question of performance, I'm sure there are plenty of users for who this would not be an issue. I tested for 2 different real pass databases, both have over 100 passwords: a FULL `pass grep <url>` takes ~5 seconds on both, and stopping after first match is of course on average much faster. 

To me an ideal solution would be something like running a full `pass grep` with `window.location.host` and sending all the results to passff as they come in (along with a loading spinner while grep is still running to indicate that the results are partial). This behavior could also of course be off by default so that only users with smaller password stores or faster computers can choose to enable it in the settings.

**Comment by tuxor1337** [comment]
You are right. When pass supports something like this, we should at least have an open discussion about supporting it in PassFF, as well (at least as an optional feature). Reopening...

**Comment by phryneas** [comment]
Just as a comparison, but my store is on the extreme side - ~650 entries and a smartcard for decryption:

```
% time pass grep test
[...]
pass grep test  3.08s user 1.45s system 1% cpu 4:18.48 total
```

So, if you add this, please don't make it a default ;)

**Comment by mbrunnen** [comment]
pass should use git grep:
```zsh
% time git --no-pager grep --no-index test
git --no-pager grep --no-index test 0.56s user 0.32s system 313% cpu 0.278 total
% time grep -r --exclude-dir=.git test
grep -r --exclude-dir=.git test 0.55s user 0.12s system 99% cpu 0.673 total
```

**Comment by Spheerys** [comment]
I'm also very interrested by finding a solution for greping the url string instead the filename for the url matching.
Is the git grep solution could be a realistic issue ?

**Comment by tuxor1337** [comment]
Feel free to make a PR and we can discuss your proposal.

**Comment by sunaurus** [comment]
I made a PR with one way of solving this: #402. Would love to get some feedback.

**Comment by tuxor1337** [comment]
I suppose this is solved by #402. Thanks, @sunaurus !
